pkg/functiontest/testing.go

```diff
@@ -0,0 +1,143 @@
+package functiontest
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/Azure/eno/pkg/function"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Scenario represents a test case for a synthesizer function.
+type Scenario[T function.Inputs] struct {
+	Name        string
+	Inputs      T
+	Assertion   Assertion[T]
+	ExpectError bool
```

Comment:
What do you think about having more expressiveness for the error expectation? Maybe an error assertion with exported defaults for "has error" and "not has error"? I'm okay with *maybe* addressing this in a follow-up?

Reply:
What do you think about just removing `ExpectError` and letting folks write their own unit tests for error conditions? I'm thinking there aren't many error conditions for most synths anyway and they should be easy to test without getting this library involved

Reply:
I can get behind that :)